be/src/exprs/runtime_filter_bank.cpp

```diff
@@ -417,6 +417,64 @@ void RuntimeFilterProbeCollector::do_evaluate(Chunk* chunk, RuntimeBloomFilterEv
         }
     }
 }
+
+void RuntimeFilterProbeCollector::do_evaluate_partial_chunk(Chunk* partial_chunk,
+                                                            RuntimeBloomFilterEvalContext& eval_context) {
+    auto& selection = eval_context.running_context.selection;
+    eval_context.running_context.use_merged_selection = false;
+    eval_context.running_context.compatibility =
+            _runtime_state->func_version() <= 3 || !_runtime_state->enable_pipeline_engine();
+
+    // since partial chunk is currently very lightweight (a bunch of const columns), use every runtime filter if possible
+    // without computing each rf's selectivity
+    for (auto kv : _descriptors) {
+        RuntimeFilterProbeDescriptor* rf_desc = kv.second;
+        const JoinRuntimeFilter* filter = rf_desc->runtime_filter();
+        if (filter == nullptr || filter->always_true()) {
+            continue;
+        }
+
+        auto* probe_expr = rf_desc->probe_expr_ctx();
+        auto* partition_by_exprs = rf_desc->partition_by_expr_contexts();
+
+        // skip if (conditions can be relaxed later)
+        //  1. multi-column runtime filter
+        //  2. runtime filter that is not slot ref, or a slot ref of a non-existent column
+        auto can_use_rf_on_partial_chunk = [&]() {
+            if (!partition_by_exprs->empty()) {
```

Comment:
```cpp
if (!partition_by_exprs->empty()) {
    auto* root=partition_by_exprs[0]->root();
    if (!root->is_slotref()) { return false;}
    auto* col_ref = down_cast<ColumnRef*>(root); 
    if (col_ref->slot_id()!=slot_id) {return false;}
} 
```

Reply:
should I also check `partition_by_exprs[1..]`? or checking `partition_by_exprs[0]` is enough?

Reply:
Update: traversed every expr to check only existent columns are referenced.